### PR TITLE
Upgrade aws-sdk version

### PR DESCRIPTION
**Pull request by pramesh94**
The aws sdk version currently is 1.11.215 which is more than 4 yrs old. This PR upgrades it to 1.11.815 which has enhanced DefaultAWSCredentialsProviderChain. This adds credentials to be received from web-identity-token file.

Refer [aws sdk documentation](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html ) for more details.

### CODE REVIEW

**Comment by yuzhu** [comment]
@LuQQiu do you see any issues with this sdk upgrade?

**Comment by LuQQiu** [comment]
@yuzhu I didn't know the difference between those two versions. The only issue I am a little concerned about is the compatibility. If the newer version doesn't break any of the compatibility, i am good with the changes

**Comment by pramesh94** [comment]
@LuQQiu the aws-sdk in the 1.11.x series are all backward compatible.. the latest version is 1.11.993. We can bump up to that also

File: underfs/s3a/pom.xml [hunk 1]
```diff
@@ -37,6 +37,10 @@
       <groupId>com.amazonaws</groupId>
       <artifactId>aws-java-sdk-s3</artifactId>
     </dependency>
+      <dependency>
+        <groupId>com.amazonaws</groupId>
+        <artifactId>aws-java-sdk-sts</artifactId>
+      </dependency>
```
Comment (apc999) [inline underfs/s3a/pom.xml:43]: nit: indent
```suggestion
    <dependency>
      <groupId>com.amazonaws</groupId>
      <artifactId>aws-java-sdk-sts</artifactId>
    </dependency>
```

**Comment by apc999** [comment]
@calvinjia  do you have any concern of this upgrade? I am willing to try it out in an edge release.

**Comment by calvinjia** [comment]
I don't see any issues with the upgrade.

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by HelloHorizon** [comment]
alluxio-bot, cherry-pick this to branch-2.5 please

**Pull request merged**